Toolkit/ArcGISToolkit/AR/ArcGISARView.swift

```diff
@@ -0,0 +1,682 @@
+    }
+    
+    /**
+     This is called after a session resumes from a pause or interruption to determine
+     whether or not the session should attempt to relocalize.
+     
+     @discussion To avoid misaligned anchors, apps may wish to attempt a relocalization after
+     a session pause or interruption. If YES is returned: the session will begin relocalizing
+     and tracking state will switch to limited with reason relocalizing. If successful, the
+     session's tracking state will return to normal. Because relocalization depends on
+     the user's location, it can run indefinitely. Apps that wish to give up on relocalization
+     may call run with `ARSessionRunOptionResetTracking` at any time.
+     @param session The session to relocalize.
+     @return Return YES to begin relocalizing.
+     */
+    @available(iOS 11.3, *)
+    public func sessionShouldAttemptRelocalization(_ session: ARSession) -> Bool {
+        if let result = sessionDelegate?.sessionShouldAttemptRelocalization?(session) {
+            return result
+        }
+        return false
+    }
+    
+    /**
+     This is called when the session outputs a new audio sample buffer.
+     
+     @param session The session being run.
+     @param audioSampleBuffer The captured audio sample buffer.
+     */
+    public func session(_ session: ARSession, didOutputAudioSampleBuffer audioSampleBuffer: CMSampleBuffer) {
+        sessionDelegate?.session?(session, didOutputAudioSampleBuffer: audioSampleBuffer)
+    }
+}
+
+// MARK: - CLLocationManagerDelegate
+extension ArcGISARView: CLLocationManagerDelegate {
+    /*
+     *  locationManager:didUpdateLocations:
+     *
+     *  Discussion:
+     *    Invoked when new locations are available.  Required for delivery of
+     *    deferred locations.  If implemented, updates will
+     *    not be delivered to locationManager:didUpdateToLocation:fromLocation:
+     *
+     *    locations is an array of CLLocation objects in chronological order.
+     */
+    public func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
+        guard let location = locations.last, location.horizontalAccuracy >= 0.0 else { return }
+        
+        if initialLocation == nil {
+            initialLocation = location
+            horizontalAccuracy = location.horizontalAccuracy
+            
+            let locationPoint = AGSPoint(x: location.coordinate.longitude,
+                                         y: location.coordinate.latitude,
+                                         z: location.altitude,
+                                         spatialReference: .wgs84())
+            let camera = AGSCamera(location: locationPoint, heading: 0.0, pitch: 0.0, roll: 0.0)
+            initialTransformationMatrix = camera.transformationMatrix
+            sceneView.setViewpointCamera(camera)
+            
+            finalizeStart()
+            print("didUpdateLocations - initialLocation...")
```

Comment:
did you want/mean to leave print statements in?

Reply:
Removed them.